### PR TITLE
Set up and update infrastructure

### PR DESCRIPTION
**Pull request by martinfleis**
So far updating docs to run with the latest sphinx and adding CI matrix for GHA.

### CODE REVIEW

**Comment by martinfleis** [comment]
Not sure how to make it run :D.

**Comment by jGaboardi** [comment]
> Not sure how to make it run :D.

From what I recall, we have two options:
1. Merge into your personal remote to make sure it runs; or
2. Live on the edge and merge here, then we can correct it.

I vote for (2) to live on the wild side.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Install pointpats
+        run: pip install .
+
+      - name: Test pointpats
+        run: |
+          pytest -v --color yes --cov pointpats --cov-append --cov-report term-missing --cov-report xml .
+
+      - uses: codecov/codecov-action@v3
+
+      - name: Generate and publish the report
```
Comment (jGaboardi) [inline .github/workflows/tests.yaml:53]: I remember seeing this in `momepy` (I think). What exactly is this doing?

**Comment by martinfleis** [comment]
Let's go wild then!

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (martinfleis) [reply .github/workflows/tests.yaml:53]: When you have a new failure in a scheduled CI run, it opens an issue so you don't miss it.

Comment (jGaboardi) [reply .github/workflows/tests.yaml:53]: ooooo. very cool!

**Comment by martinfleis** [comment]
That did not do the trick https://github.com/pysal/pointpats/actions

**Comment by jGaboardi** [comment]
@martinfleis Did you update the target?

**Pull request merged**